### ISSUE TITLE
GoPro as dashcam : Issue on Motion detection trigger in a scheduled interval

### ISSUE THREAD

**Issue by esiremalaysia** [esiremalaysia/gopro#1]
## Introducing GoPro as a dashcam

After several tests and stress testing, the full needed features are finally working mainly on H12 and partially on H11

### Camera compatibility and firmware : 
- [x]  _tested on H12, [firmware v1.12.70](https://github.com/esiremalaysia/gopro/blob/main/labs/H12/LABS_HERO12_01_12_70.zip), September 29, 2023_
- [x]  _tested on H11, [firmware v2.12.70](https://github.com/esiremalaysia/gopro/blob/main/labs/H11/LABS_HERO11_02_30_70.zip), August 2023_
- [x]  _tested on H11 mini, firmware v2.30.70, May 2023_

### Dashcam Mode while driving (day/night)
- **Use case explained and scripts :** The script boot automatically upon USB power, it will then wait until it detects acceleration before starting recording. The script will also create space as your SD card fills. It will remain recording until USB power is removed but it will continue to record while it detects any physical camera motion, but stopped 10 seconds after all motion has ceased. The camera will remain on for a further 20 seconds just in case USB power is returned, but otherwise it will shutdown, ready to automatically power up again and start all over. 
   - ✅ 1. All the above (only on H11) :`!MWAKE=2!MTUSB=1!MBOOT="!Ldcam"!SAVEdcam=<u0!R97!C16<a0.05!R9=L 1>L99!C16 =L0=Bt:B>u0<r0!S =AB>u0=AB>a0.05=AB=B-A>B10>r0<u0!EB30!O!R18!B2`
   - ✅ 2. Start on power USB with no motion detection at the end (H11/H12) : `!MBOOT="!Lbt"!SAVEbt="Dashcam"+>u0<r0!C16+!S+=A1>A1<u0>r0<m1=A+1+!1N~=A1>A6>r0!E+!1O!R10`
- _**Additional thing :** script is clever and detects that it has been started without USB power and behave like a regular GoPro. So you can use it as a dashcam, take it out, uses a regular GoPro without ever having to re-scan a QR code_ 
- related issue : #484, #396
- demo [here](https://www.youtube.com/watch?v=T87RekjaVA4)

### Dashcam Mode when parked (day)
   _As gopro cameras had bad low light performance until H12 (01/10/23), it is not interesting to configure it for night time, but if there is a good outdoor lighting, just skip intervals condition and directly use the QR code n°2_

- **All use cases explained and scripts (H12) :** 
  - ✅ 1. Start Motion Detection continuously :  `!SM4H2!R`
  - ✅ 2. Start Motion Detection + Hindsight (15s window) continuously : `hS1!1N!SM4H2hS0!R`
  - ✅ 3. Start Motion Detection continuously on a scheduled time (from 8:00AM to 20:00AM) : `<20:00!SM4H2~!08:00R!R`
  - ✅ 4. Start Motion Detection continuously on a scheduled time (from 8:00AM to 20:00AM) + Hindsight (15s window) : `<20:00hS1+!1N+!SM4H2+hS0~!08:00N!R`
  - ✅ 5. Start Motion Detection continuously on a scheduled time **with a precise starting time** (from 8:00AM to 20:00AM) + Hindsight (15s window) : `>05:00<20:00hS1+!1N+!SM4H2+hS0~!08:00N<05:00!08:00N!R`
  - 🔄 6. All the above (motion + hindsight + daylight schedule ) + upload on the cloud : todo

**note:** The advantage of script 5 over 4 is that, script 5 had more control on time loops and when script must be executed. For example, if you run the script at 3AM in the morning, script 4 will enter the dashcam script immediately and start processing due to his single condition loop whereas script 5 will shutdown and run the dashcam script only if the time is between 5AM and 20AM due to his stacked conditionals loop. Plus, one thing,  H11 is not stable as H12 when testing Dashcam Mode when it's parked.

### Enhancement:

**- Battery**
The minor issue is power delivery. While doing these tasks I want to loop the needed script eternally without any lagging issue. The problem is that plugged to a power bank (limited power) or an adapter (continuous power) to the H12 it has some weird behavior like after long time exposure to the script (> 3 days): 

- black screen, 
- overheat and shut off, 
- screen blocked, cannot interact with
- or the loop still continues with Hindsight but without the Motion Detection (rare case), 
- often the need to take off and put back the battery to initialize the camera due to these previous bugs

I assume that those issues are not related with the dashcam script thing but with the overall H12 architecture, capabilities and limits. But overall, it's functional now with good behavior for 3 days straight and continuously. Thanks @dnewman-gpsw 

**- Update**
This script will be tested on newer firmware versions of Labs on H12 in the future, to ensure his maintenance over time. Actual firmware to test : [HERO12 Black](https://github.com/esiremalaysia/gopro/blob/main/labs/H12/LABS_HERO12_02_20_70.zip) - v2.20.70, March 2024

### Logs debugging 
Improve logging verbosity (Debug Level 1)
- `!MDBGL=1`

### Reset Labs
- `!RESET`
